internal/cmd/kafka/command_cluster_update.go

```diff
@@ -45,18 +49,22 @@ func (c *clusterCommand) newUpdateCommand(cfg *v1.Config) *cobra.Command {
 }
 
 func (c *clusterCommand) update(cmd *cobra.Command, args []string, prompt form.Prompt) error {
-	if !cmd.Flags().Changed("name") && !cmd.Flags().Changed("cku") {
-		return errors.New(errors.NameOrCKUFlagErrorMsg)
+	if !(cmd.Flags().Changed("name") || cmd.Flags().Changed("availability") || cmd.Flags().Changed("type") || cmd.Flags().Changed("cku")) {
+		return errors.New("must specify one of `--name`, `--availability`, `--type`, or `--cku`")
 	}
 
-	clusterID := args[0]
-	currentCluster, _, err := c.V2Client.DescribeKafkaCluster(clusterID, c.EnvironmentId())
+	clusterId := args[0]
+	currentCluster, _, err := c.V2Client.DescribeKafkaCluster(clusterId, c.EnvironmentId())
 	if err != nil {
-		return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.KafkaClusterNotFoundErrorMsg, clusterID), errors.ChooseRightEnvironmentSuggestions)
+		return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.KafkaClusterNotFoundErrorMsg, clusterId), errors.ChooseRightEnvironmentSuggestions)
+	}
+
+	if currentCluster.Status.GetPhase() == ccloudv2.StatusProvisioning {
+		return errors.New(errors.KafkaClusterProvisioningErrorMsg)
```

Comment:
Just for my understanding, is this just ensuring that if the cluster is still provisioning, we don't make any updates?